cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java

```diff
@@ -130,21 +124,17 @@ public static CruiseControl fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup ver
                     kafkaAssembly.getMetadata().getName(),
                     Labels.fromResource(kafkaAssembly).withKind(kafkaAssembly.getKind()));
 
-        CruiseControlSpec spec  = kafkaAssembly.getSpec().getCruiseControl();
+        KafkaSpec kafkaSpec  = kafkaAssembly.getSpec();
+        CruiseControlSpec spec  = kafkaSpec.getCruiseControl();
 
         if (spec != null) {
             cruiseControl.isDeployed = true;
 
-            cruiseControl.setReplicas(spec != null && spec.getReplicas() > 0 ? spec.getReplicas() : DEFAULT_REPLICAS);
+            cruiseControl.setReplicas(spec.getReplicas() > 0 ? spec.getReplicas() : DEFAULT_REPLICAS);
```

Comment:
so we are allowing minimum 0 in the spec but then forced to `DEFAULT_REPLICAS` which is 1 if someone set `replicas` 0?

Reply:
Forgot to update this!